backends-common/rabbitmq/src/test/java/org/apache/james/backends/rabbitmq/RabbitMQTest.java

```diff
@@ -327,6 +330,137 @@ void rabbitMQShouldSupportTheExclusiveWorkQueueCase() throws Exception {
                 assertThat(fallbackConsumer.getConsumedMessages()).contains(1, 2).doesNotContain(0);
             }
 
+            @Test
+            void rabbitMQShouldDeliverMessageToSingleActiveConsumer() throws Exception {
+                channel1.exchangeDeclare(EXCHANGE_NAME, "direct", DURABLE);
+                channel1.queueDeclare(WORK_QUEUE, DURABLE, !EXCLUSIVE, !AUTO_DELETE, Constants.WITH_SINGLE_ACTIVE_CONSUMER);
+                channel1.queueBind(WORK_QUEUE, EXCHANGE_NAME, ROUTING_KEY);
+
+                IntStream.range(0, 10)
+                    .mapToObj(String::valueOf)
+                    .map(RabbitMQTest.this::asBytes)
+                    .forEach(Throwing.consumer(
+                        bytes -> channel1.basicPublish(EXCHANGE_NAME, ROUTING_KEY, NO_PROPERTIES, bytes)));
+
+                channel2.basicQos(1);
+                channel3.basicQos(1);
+
+                AtomicInteger firstRegisteredConsumerMessageCount = new AtomicInteger(0);
+                AtomicInteger secondRegisteredConsumerMessageCount = new AtomicInteger(0);
+
+                String firstRegisteredConsumer = "firstRegisteredConsumer";
+                ImmutableMap<String, Object> arguments = ImmutableMap.of();
+                channel2.basicConsume(WORK_QUEUE, !AUTO_ACK, firstRegisteredConsumer, !NO_LOCAL, !EXCLUSIVE, arguments,
```

Comment:
can you extract it please?

Reply:
it's pretty straightforward, we could extract empty arguments to a constant but extracting the call to basicConsume would not really help understanding.
Or maybe you want the lambda to be extracted?

Reply:
It's bulky, I prefer having the whole thing defined in a dedicated method with a good name, which will allow me to under *what*, and eventually, as needed, *how*.

Reply:
I extracted the lambda